NuGet.Services.Dashboard.Operations/Infrastructure/CommandLineParser.cs

```diff
@@ -70,7 +72,16 @@ public void ExtractOptions(ICommand command, IEnumerator<string> argsEnumerator)
                     throw new CommandLineException(TaskResources.MissingOptionValueError, option);
                 }
 
-                AssignValue(command, propInfo, option, value);
+                string newValue;
+
+                if (SecretConfigurationManager.TryInjectSecret(value, out newValue))
```

Comment:
Correct me if my understanding here is incorrect, but does this mean that for every command line arg, we attempt to fetch a secret from KeyVault, and if it fails then we assume it was not a secret? That sounds very inefficient. From personal testing, accessing KeyVault takes a non-trivial amount of time. I think we should delimit the things that ARE secrets with a symbol (`$$` perhaps) and then check for it and parse it out here. Additionally, that will let us know if something we know is a secret fails to load, rather than just assuming that it was not a secret to begin with.